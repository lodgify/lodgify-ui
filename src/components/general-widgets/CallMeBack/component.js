import React from 'react';
import PropTypes from 'prop-types';

import {
  CALL_ME_BACK,
  DATE,
  EMAIL,
  NAME,
  NOTES,
  PHONE,
  PROPERTY,
  SEND,
  TIME,
} from 'utils/default-strings';
import { Dropdown } from 'inputs/Dropdown';
import { Form } from 'collections/Form';
import { ICON_NAMES } from 'elements/Icon';
import { InputGroup } from 'collections/InputGroup';
import { PhoneInput } from 'inputs/PhoneInput';
import { SingleDatePicker } from 'inputs/SingleDatePicker';
import { TextArea } from 'inputs/TextArea';
import { TextInput } from 'inputs/TextInput';

/**
 * The standard widget for a user to request a call back.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  dateInputPlaceholder,
  emailInputLabel,
  errorMessage,
  headingText,
  nameInputLabel,
  notesInputLabel,
  onSubmit,
  phoneInputLabel,
  propertyInputLabel,
  propertyOptions,
  submitButtonText,
  successMessage,
  timeInputLabel,
  timeOptions,
  validation,
}) => (
  <Form
    autoComplete="off"
    errorMessage={errorMessage}
    headingText={headingText}
    onSubmit={onSubmit}
    submitButtonText={submitButtonText}
    successMessage={successMessage}
    validation={validation}
  >
    <InputGroup>
      <TextInput autoComplete="name" label={nameInputLabel} name="name" />
      <PhoneInput autoComplete="tel" label={phoneInputLabel} name="phone" />
    </InputGroup>
    <TextInput autoComplete="email" label={emailInputLabel} name="email" />
    <InputGroup>
      <SingleDatePicker name="date" placeholderText={dateInputPlaceholder} />
      <Dropdown
        icon={ICON_NAMES.CLOCK}
        label={timeInputLabel}
        name="time"
        options={timeOptions}
      />
    </InputGroup>
    <Dropdown
      label={propertyInputLabel}
      name="property"
      options={propertyOptions}
    />
    <TextArea label={notesInputLabel} name="notes" />
  </Form>
);

Component.displayName = 'CallMeBack';

Component.defaultProps = {
  dateInputPlaceholder: DATE,
  emailInputLabel: EMAIL,
  errorMessage: '',
  headingText: CALL_ME_BACK,
  nameInputLabel: NAME,
  notesInputLabel: NOTES,
  onSubmit: Function.prototype,
  phoneInputLabel: PHONE,
  propertyInputLabel: PROPERTY,
  submitButtonText: SEND,
  successMessage: '',
  timeInputLabel: TIME,
  validation: {},
};

Component.propTypes = {
  /** The placeholder for the date input. */
  dateInputPlaceholder: PropTypes.string,
  /** The label for the email input. */
  emailInputLabel: PropTypes.string,
  /** The message to display when the form has an error. */
  errorMessage: PropTypes.string,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The label for the name input. */
  nameInputLabel: PropTypes.string,
  /** The label for the notes input. */
  notesInputLabel: PropTypes.string,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The label for the phone input */
  phoneInputLabel: PropTypes.string,
  /** The label for the property input. */
  propertyInputLabel: PropTypes.string,
  /** The options which the user can select for the property field. */
  propertyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** The form submit button text */
  submitButtonText: PropTypes.string,
  /** The message to display when the form is successful. */
  successMessage: PropTypes.string,
  /** The label for the time input. */
  timeInputLabel: PropTypes.string,
  /** The options which the user can select for the time field. */
  timeOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** Settings for validating inputs. Each value should match [the shape documented in `Form`](https://lodgify.github.io/lodgify-ui/#/Collections/Form) */
  validation: PropTypes.shape({
    date: PropTypes.object,
    email: PropTypes.object,
    name: PropTypes.object,
    notes: PropTypes.object,
    phone: PropTypes.object,
    property: PropTypes.object,
    time: PropTypes.object,
  }),
};
