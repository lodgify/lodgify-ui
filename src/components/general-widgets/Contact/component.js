import React from 'react';
import PropTypes from 'prop-types';

import { Form } from 'collections/Form';
import { InputGroup } from 'collections/InputGroup';
import { TextInput } from 'inputs/TextInput';
import { PhoneInput } from 'inputs/PhoneInput';
import { NumberInput } from 'inputs/NumberInput';
import { DateRangePicker } from 'inputs/DateRangePicker';
import { TextArea } from 'inputs/TextArea';
import { CaptchaInput } from 'inputs/CaptchaInput';
import { Dropdown } from 'inputs/Dropdown';
import {
  ARRIVAL,
  CONTACT,
  COMMENTS,
  DEPARTURE,
  EMAIL,
  GUESTS,
  NAME,
  PHONE,
  PROPERTY,
  ROOM,
  SECURITY_CODE,
  SEND,
} from 'utils/default-strings';

/**
 * The standard widget for a user contact an owner.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  arrivalDateInputLabel,
  captchaInputImage,
  captchaInputLabel,
  commentsInputLabel,
  dateRangePickerLocaleCode,
  departureDateInputLabel,
  emailInputLabel,
  errorMessage,
  guestsInputLabel,
  headingText,
  nameInputLabel,
  onChangeProperty,
  onSubmit,
  phoneInputLabel,
  propertyInputLabel,
  propertyOptions,
  roomInputLabel,
  roomOptions,
  submitButtonText,
  successMessage,
  validation,
}) => (
  <Form
    autoComplete="off"
    errorMessage={errorMessage}
    headingText={headingText}
    onSubmit={onSubmit}
    submitButtonText={submitButtonText}
    successMessage={successMessage}
    validation={validation}
  >
    <InputGroup>
      <TextInput autoComplete="name" label={nameInputLabel} name="name" />
      <PhoneInput autoComplete="tel" label={phoneInputLabel} name="phone" />
    </InputGroup>
    <TextInput autoComplete="email" label={emailInputLabel} name="email" />
    <InputGroup>
      <DateRangePicker
        endDatePlaceholderText={departureDateInputLabel}
        localeCode={dateRangePickerLocaleCode}
        name="dates"
        startDatePlaceholderText={arrivalDateInputLabel}
        width="eight"
      />
      <NumberInput
        label={guestsInputLabel}
        min={1}
        name="guests"
        width="four"
      />
    </InputGroup>
    <TextArea label={commentsInputLabel} name="comments" />
    {(roomOptions || propertyOptions) && (
      <InputGroup>
        {propertyOptions && (
          <Dropdown
            label={propertyInputLabel}
            name="property"
            onChange={onChangeProperty}
            options={propertyOptions}
          />
        )}
        {roomOptions && (
          <Dropdown label={roomInputLabel} name="room" options={roomOptions} />
        )}
      </InputGroup>
    )}
    <CaptchaInput
      image={captchaInputImage}
      label={captchaInputLabel}
      name="captcha"
    />
  </Form>
);

Component.displayName = 'Contact';

Component.defaultProps = {
  arrivalDateInputLabel: ARRIVAL,
  captchaInputLabel: SECURITY_CODE,
  commentsInputLabel: COMMENTS,
  dateRangePickerLocaleCode: undefined,
  departureDateInputLabel: DEPARTURE,
  emailInputLabel: EMAIL,
  errorMessage: '',
  guestsInputLabel: GUESTS,
  headingText: CONTACT,
  nameInputLabel: NAME,
  onChangeProperty: Function.prototype,
  onSubmit: Function.prototype,
  phoneInputLabel: PHONE,
  propertyInputLabel: PROPERTY,
  propertyOptions: null,
  roomInputLabel: ROOM,
  roomOptions: null,
  submitButtonText: SEND,
  successMessage: '',
  validation: {},
};

Component.propTypes = {
  /** The label for the arrival date input.*/
  arrivalDateInputLabel: PropTypes.string,
  /** The source url for the image to display. */
  captchaInputImage: PropTypes.string.isRequired,
  /** The label for the captcha input.*/
  captchaInputLabel: PropTypes.string,
  /** The label for the comments input.*/
  commentsInputLabel: PropTypes.string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year in the date range picker. */
  dateRangePickerLocaleCode: PropTypes.string,
  /** The label for the departure date input.*/
  departureDateInputLabel: PropTypes.string,
  /** The label for the email input.*/
  emailInputLabel: PropTypes.string,
  /** The message to display when the form has an error. */
  errorMessage: PropTypes.string,
  /** The label for the guests input.*/
  guestsInputLabel: PropTypes.string,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The label for the name input.*/
  nameInputLabel: PropTypes.string,
  /** The function called when the property dropdown is changed.
   *  @param {String}        name - The name of the property dropdown field.
   *  @param {String|Number} value - The value of the property dropdown after the change
   */
  onChangeProperty: PropTypes.func,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The label for the phone input.*/
  phoneInputLabel: PropTypes.string,
  /** The label for the property input.*/
  propertyInputLabel: PropTypes.string,
  /** The options which the user can select for the property field. */
  propertyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The indent level of a property option. One of: 0, 1, 2, 3, 4, 5 */
      indent: PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The label for the room input.*/
  roomInputLabel: PropTypes.string,
  /** The options which the user can select for the room field. */
  roomOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The text to display on the submit button. */
  submitButtonText: PropTypes.string,
  /** The message to display when the form is successful. */
  successMessage: PropTypes.string,
  /** Settings for validating inputs. Each value should match [the shape documented in `Form`](https://lodgify.github.io/lodgify-ui/#/Collections/Form) */
  validation: PropTypes.shape({
    captcha: PropTypes.object,
    comments: PropTypes.object,
    dates: PropTypes.object,
    email: PropTypes.object,
    guests: PropTypes.object,
    name: PropTypes.object,
    phone: PropTypes.object,
    property: PropTypes.object,
    room: PropTypes.object,
  }),
};
