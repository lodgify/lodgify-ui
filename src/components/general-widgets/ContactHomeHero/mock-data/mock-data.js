export const navigationItems = [
  { text: 'Home', href: '/' },
  {
    text: 'All properties',
    subItems: [
      {
        href: '/la-casa-viva',
        text: 'La Casa Viva',
      },
      {
        href: '/la-casa-muerta',
        text: 'La Casa Muerta',
      },
      {
        href: '/the-white-lodge',
        text: 'The White Lodge',
      },
      {
        href: '/the-black-lodge',
        text: 'The Black Lodge',
      },
    ],
  },
  {
    text: 'LOL',
    subItems: [
      {
        href: '/la-casa-viva',
        text: 'La Casa Viva',
      },
      {
        href: '/la-casa-muerta',
        text: 'La Casa Muerta',
      },
      {
        href: '/the-white-lodge',
        text: 'The White Lodge',
      },
      {
        href: '/the-black-lodge',
        text: 'The Black Lodge',
      },
    ],
  },
];

export const currencyOptions = [
  { text: 'EUR', value: 'EUR', label: 'Euro' },
  { text: 'USD', value: 'USD', label: 'US Dollar' },
  { text: 'JPY', value: 'JPY', label: 'Japanese Yen' },
];

export const languageOptions = [
  { text: 'English', value: 'en' },
  { text: 'German', value: 'de' },
  { text: 'Japanese', value: 'jp' },
  { text: 'Italian', value: 'it' },
];

export const backgroundImageUrl =
  'https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg';
