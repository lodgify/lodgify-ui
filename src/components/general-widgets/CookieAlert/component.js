import React from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'elements/Modal';
import { FlexContainer } from 'layout/FlexContainer';
import { Button } from 'elements/Button';
import { Divider } from 'elements/Divider';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import { Paragraph } from 'typography/Paragraph';
import { withResponsive } from 'utils/with-responsive';
import { Link } from 'elements/Link';

import { getFormMarkup } from './utils/getFormMarkup';

const Component = ({
  buttonText,
  isOpen,
  isUserOnMobile,
  linkText,
  linkUrl,
  onAccept,
  text,
}) => {
  if (!isOpen) return false;
  return isUserOnMobile ? (
    <Modal hasCloseIcon={false} hasRoundedCorners isOpen>
      <HorizontalGutters>
        <FlexContainer
          alignItems="center"
          flexDirection="column"
          justifyContent="center"
        >
          <Divider />
          <Paragraph>{text}</Paragraph>
          <Link href={linkUrl}>{linkText}</Link>
          <Divider />
          <Button onClick={onAccept}>{buttonText}</Button> <Divider />
        </FlexContainer>
      </HorizontalGutters>
    </Modal>
  ) : (
    <div className="cookies-form">
      {getFormMarkup(buttonText, linkText, linkUrl, onAccept, text)}
    </div>
  );
};

Component.displayName = 'CookieAlert';

Component.defaultProps = {
  buttonText: 'Accept',
  isOpen: false,
  linkText: 'Learn More',
  linkUrl: '/',
};

Component.propTypes = {
  /** The text to display on the button inside the component. */
  buttonText: PropTypes.string,
  /** Is the modal open. */
  isOpen: PropTypes.bool,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
  /** The text to display as a link. */
  linkText: PropTypes.string,
  /** The url attached to the link. */
  linkUrl: PropTypes.string,
  /** The on accept handler for the call to action. */
  onAccept: PropTypes.func.isRequired,
  /** The text to display in the component. */
  text: PropTypes.string.isRequired,
  /** The element to be clicked to display the modal. */
};

export const ComponentWithResponsive = withResponsive(Component);
