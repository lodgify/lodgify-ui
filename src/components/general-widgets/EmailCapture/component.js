import 'semantic-ui-styles/message.less';
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Message } from 'semantic-ui-react';

import { withResponsive } from 'utils/with-responsive';
import { Grid } from 'layout/Grid';
import { GridRow } from 'layout/GridRow';
import { GridColumn } from 'layout/GridColumn';
import { Heading } from 'typography/Heading';
import { ShowOn } from 'layout/ShowOn';
import { TextInput } from 'inputs/TextInput';
import { Button } from 'elements/Button';
import { Divider } from 'elements/Divider';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import {
  ACCEPT_PRIVACY_POLICY,
  SUBSCRIBE,
  SUBSCRIBE_TO_OUR_NEWSLETTER,
  YOUR_EMAIL,
} from 'utils/default-strings';

import { getPrivacyCheckboxMarkup } from './utils/getPrivacyCheckboxMarkup';

/**
 * An email capture displays a form inviting a user to submit an email address.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
const Component = ({
  buttonText,
  emailInputError,
  emailInputLabel,
  emailInputValue,
  errorMessage,
  headingText,
  isPrivacyConsentInputChecked,
  isPrivacyConsentRequired,
  isUserOnMobile,
  onChangeEmailInput,
  onClickButton,
  onClickPrivacyConsentInput,
  privacyConsentInputError,
  privacyConsentLabelLinkText,
  privacyConsentLabelLinkUrl,
  privacyConsentLabelText,
  successMessage,
}) => (
  <HorizontalGutters>
    <Divider />
    <Grid>
      <GridRow verticalAlign="middle">
        {!errorMessage ? (
          <Fragment>
            <GridColumn computer={5} mobile={12} tablet={12} textAlign="center">
              <Heading hasMargin={false} size="small">
                {headingText}
              </Heading>
              <ShowOn
                mobile
                parent={Divider}
                parentProps={{ size: 'small' }}
                tablet
              />
            </GridColumn>
            <GridColumn computer={4} mobile={12} tablet={7}>
              <TextInput
                error={emailInputError}
                isFluid
                label={emailInputLabel}
                onChange={onChangeEmailInput}
                value={emailInputValue}
              />
              <ShowOn mobile parent={Divider} parentProps={{ size: 'small' }} />
            </GridColumn>
            <GridColumn computer={3} mobile={12} tablet={5}>
              {isUserOnMobile && isPrivacyConsentRequired && (
                <Fragment>
                  {getPrivacyCheckboxMarkup(
                    privacyConsentInputError,
                    isPrivacyConsentInputChecked,
                    privacyConsentLabelText,
                    privacyConsentLabelLinkUrl,
                    privacyConsentLabelLinkText,
                    onClickPrivacyConsentInput
                  )}
                  <Divider />
                </Fragment>
              )}
              {!successMessage ? (
                <Button isFluid isRounded onClick={onClickButton}>
                  {buttonText}
                </Button>
              ) : (
                <Message content={successMessage} success />
              )}
            </GridColumn>
          </Fragment>
        ) : (
          <GridColumn width={12}>
            <Message content={errorMessage} error />
          </GridColumn>
        )}
      </GridRow>
      {!errorMessage && !isUserOnMobile && isPrivacyConsentRequired && (
        <GridRow>
          <GridColumn computer={5} />
          <GridColumn computer={4} mobile={12} tablet={12}>
            {getPrivacyCheckboxMarkup(
              privacyConsentInputError,
              isPrivacyConsentInputChecked,
              privacyConsentLabelText,
              privacyConsentLabelLinkUrl,
              privacyConsentLabelLinkText,
              onClickPrivacyConsentInput
            )}
          </GridColumn>
        </GridRow>
      )}
    </Grid>
    <Divider />
  </HorizontalGutters>
);

export const ComponentWithResponsive = withResponsive(Component);

Component.displayName = 'EmailCapture';

Component.defaultProps = {
  buttonText: SUBSCRIBE,
  emailInputError: false,
  emailInputLabel: YOUR_EMAIL,
  emailInputValue: undefined,
  errorMessage: null,
  headingText: SUBSCRIBE_TO_OUR_NEWSLETTER,
  isPrivacyConsentInputChecked: false,
  isPrivacyConsentRequired: false,
  onChangeEmailInput: Function.prototype,
  onClickPrivacyConsentInput: Function.prototype,
  onClickButton: Function.prototype,
  privacyConsentInputError: undefined,
  privacyConsentLabelLinkText: undefined,
  privacyConsentLabelLinkUrl: undefined,
  privacyConsentLabelText: ACCEPT_PRIVACY_POLICY,
  successMessage: null,
};

Component.propTypes = {
  /** The text to display on the button. */
  buttonText: PropTypes.string,
  /** An error message to display in place of the component. */
  emailInputError: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** The visible label for the text input. */
  emailInputLabel: PropTypes.string,
  /** The value of the text input. */
  emailInputValue: PropTypes.string,
  /** Is the text input in an error state. */
  errorMessage: PropTypes.string,
  /** The text to display as a heading. */
  headingText: PropTypes.string,
  /** The value of the privacy consent checkbox. */
  isPrivacyConsentInputChecked: PropTypes.bool,
  /** Displays a privacy consent checkbox. */
  isPrivacyConsentRequired: PropTypes.bool,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
  /** The function called when the text input is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeEmailInput: PropTypes.func,
  /** The function called when the button is clicked.
   *  @param {String} name
   *  @param {String} value
   */
  onClickButton: PropTypes.func,
  /** The function called when the privacy consent checkbox is clicked.
   *  @param {Object} event
   */
  onClickPrivacyConsentInput: PropTypes.func,
  /** The text to display in the privacy consent error message. */
  privacyConsentInputError: PropTypes.string,
  /** The text to display as the link next to the privacy consent checkbox. */
  privacyConsentLabelLinkText: PropTypes.string,
  /** The location of the link next to the privacy consent checkbox. */
  privacyConsentLabelLinkUrl: PropTypes.string,
  /** The text to display next to the privacy consent checkbox. */
  privacyConsentLabelText: PropTypes.node,
  /** A success message to display in place of the button. */
  successMessage: PropTypes.string,
};
