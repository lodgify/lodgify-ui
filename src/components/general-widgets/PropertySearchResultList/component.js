import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import { Divider } from 'elements/Divider';
import { TextPlaceholder } from 'elements/TextPlaceholder';
import { Heading } from 'typography/Heading';
import { FlexContainer } from 'layout/FlexContainer';
import { PropertySearchResult } from 'general-widgets/PropertySearchResult';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { RESULTS } from 'utils/default-strings';

import { PLACEHOLDERS } from './constants';

/**
 * The standard widget for displaying a list of property search results.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  isShowingPlaceholder,
  propertySearchResults,
  resultsCountText,
}) =>
  isShowingPlaceholder ? (
    <Fragment>
      <Divider size="small" />
      <TextPlaceholder length="short" />
      <Divider size="small" />
      <FlexContainer flexWrap="wrap" justifyContent="center">
        {PLACEHOLDERS.map((propertySearchResult, index) => (
          <PropertySearchResult
            isShowingPlaceholder
            key={buildKeyFromStrings(
              propertySearchResult.placeholderName,
              index
            )}
            {...propertySearchResult}
          />
        ))}
      </FlexContainer>
    </Fragment>
  ) : (
    <Fragment>
      <Heading size="small">
        {`${propertySearchResults.length} ${resultsCountText}`}
      </Heading>
      <FlexContainer flexWrap="wrap" justifyContent="center">
        {propertySearchResults.map((propertySearchResult, index) => (
          <PropertySearchResult
            key={buildKeyFromStrings(propertySearchResult.propertyName, index)}
            {...propertySearchResult}
          />
        ))}
      </FlexContainer>
    </Fragment>
  );

Component.displayName = 'PropertySearchResultList';

Component.defaultProps = {
  propertySearchResults: [],
  resultsCountText: RESULTS,
  isShowingPlaceholder: false,
};

Component.propTypes = {
  /** Is the component showing placeholders to reserve space for content which will appear. */
  isShowingPlaceholder: PropTypes.bool,
  /** An array of [`PropertySearchResult`](#/PropertySearchResult) props objects. */
  propertySearchResults: PropTypes.arrayOf(PropTypes.object),
  /** The text to display alongside the results count. */
  resultsCountText: PropTypes.string,
};
