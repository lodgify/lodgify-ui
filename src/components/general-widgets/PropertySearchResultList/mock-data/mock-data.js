export const propertySearchResults = [
  {
    bathsNumber: 2,
    bathsText: 'baths',
    bedroomsNumber: 2,
    bedroomsText: 'bedrooms',
    bedsNumber: 2,
    bedsText: 'beds',
    guestsNumber: 2,
    guestsText: 'guests',
    imageUrl:
      'https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg',
    pricePerPeriod: '$28000',
    periodText: 'per month',
    propertyAmenities: ['Pool', 'Wifi', 'Washer', 'Kitchen'],
    propertyName: 'The Cat House',
    propertyType: 'Bed and breakfast',
    propertyUrl: '/',
    ratingNumber: 4.7,
  },
  {
    bathsNumber: 1,
    bathsText: 'baths',
    bedroomsNumber: 2,
    bedroomsText: 'bedrooms',
    bedsNumber: 3,
    bedsText: 'beds',
    guestsNumber: 4,
    guestsText: 'guests',
    imageUrl:
      'https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg',
    pricePerPeriod: '$1280',
    propertyAmenities: ['Pool', 'Wifi', 'Washer', 'Kitchen'],
    propertyName: 'Space Hub A-675',
    propertyType: 'Space Station',
    propertyUrl: '/',
    ratingNumber: 4.7,
  },
];
