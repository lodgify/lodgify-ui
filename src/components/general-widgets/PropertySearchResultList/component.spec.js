import React from 'react';
import { mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component as PropertySearchResultList } from './component';
import { propertySearchResults } from './mock-data/mock-data';

const getPropertySearchResultList = otherProps =>
  mount(
    <PropertySearchResultList
      propertySearchResults={propertySearchResults}
      {...otherProps}
    />
  );

describe('<PropertySearchResultList />', () => {
  describe('by default', () => {
    it('should render the right structure', () => {
      const actual = getPropertySearchResultList();

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.isShowingPlaceholder` is true', () => {
    it('should render the right structure', () => {
      const actual = getPropertySearchResultList({
        isShowingPlaceholder: true,
        propertySearchResults: [],
      });

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.resultsCountText` is passed', () => {
    it('should render the right structure', () => {
      const resultsCountText = 'My god I love Cliff Richard';
      const actual = getPropertySearchResultList({ resultsCountText });

      expect(actual).toMatchSnapshot();
    });
  });

  it('should have `displayName` `PropertySearchResultList`', () => {
    expectComponentToHaveDisplayName(
      PropertySearchResultList,
      'PropertySearchResultList'
    );
  });
});
