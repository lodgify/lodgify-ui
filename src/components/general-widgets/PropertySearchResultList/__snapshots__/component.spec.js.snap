// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PropertySearchResultList /> by default should render the right structure 1`] = `
<PropertySearchResultList
  isShowingPlaceholder={false}
  propertySearchResults={
    Array [
      Object {},
      Object {},
    ]
  }
  resultsCountText="results"
>
  <Heading
    className={null}
    hasMargin={true}
    isColorInverted={false}
    size="small"
    textAlign={null}
  >
    <Header
      as="h4"
      className=""
      inverted={false}
      textAlign={null}
    >
      <h4
        className="ui header"
      >
        2 results
      </h4>
    </Header>
  </Heading>
  <FlexContainer
    alignContent={null}
    alignItems={null}
    flexDirection={null}
    flexWrap="wrap"
    justifyContent={null}
  >
    <div
      className="flex-container"
      style={
        Object {
          "alignContent": null,
          "alignItems": null,
          "display": "flex",
          "flexDirection": null,
          "flexGrow": "1",
          "flexWrap": "wrap",
          "height": "100%",
          "justifyContent": null,
        }
      }
    >
      <PropertySearchResult
        key="0"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
      <PropertySearchResult
        key="1"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
    </div>
  </FlexContainer>
</PropertySearchResultList>
`;

exports[`<PropertySearchResultList /> if \`props.isShowingPlaceholder\` is true should render the right structure 1`] = `
<PropertySearchResultList
  isShowingPlaceholder={true}
  propertySearchResults={
    Array [
      Object {},
      Object {},
    ]
  }
  resultsCountText="results"
>
  <Divider
    className=""
    hasLine={false}
    size="small"
  >
    <Divider
      className="is-size-small"
      hidden={true}
      section={false}
    >
      <div
        className="ui hidden divider is-size-small"
      />
    </Divider>
  </Divider>
  <TextPlaceholder
    isFluid={true}
    length="short"
  >
    <div
      className="placeholder text short fluid"
    />
  </TextPlaceholder>
  <Divider
    className=""
    hasLine={false}
    size="small"
  >
    <Divider
      className="is-size-small"
      hidden={true}
      section={false}
    >
      <div
        className="ui hidden divider is-size-small"
      />
    </Divider>
  </Divider>
  <FlexContainer
    alignContent={null}
    alignItems={null}
    flexDirection={null}
    flexWrap="wrap"
    justifyContent={null}
  >
    <div
      className="flex-container"
      style={
        Object {
          "alignContent": null,
          "alignItems": null,
          "display": "flex",
          "flexDirection": null,
          "flexGrow": "1",
          "flexWrap": "wrap",
          "height": "100%",
          "justifyContent": null,
        }
      }
    >
      <PropertySearchResult
        isShowingPlaceholder={true}
        key="placeholder0"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
      <PropertySearchResult
        isShowingPlaceholder={true}
        key="placeholder1"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
      <PropertySearchResult
        isShowingPlaceholder={true}
        key="placeholder2"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
      <PropertySearchResult
        isShowingPlaceholder={true}
        key="placeholder3"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
      <PropertySearchResult
        isShowingPlaceholder={true}
        key="placeholder4"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
      <PropertySearchResult
        isShowingPlaceholder={true}
        key="placeholder5"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
    </div>
  </FlexContainer>
</PropertySearchResultList>
`;

exports[`<PropertySearchResultList /> if \`props.resultsCountText\` is passed should render the right structure 1`] = `
<PropertySearchResultList
  isShowingPlaceholder={false}
  propertySearchResults={
    Array [
      Object {},
      Object {},
    ]
  }
  resultsCountText="My god I love Cliff Richard"
>
  <Heading
    className={null}
    hasMargin={true}
    isColorInverted={false}
    size="small"
    textAlign={null}
  >
    <Header
      as="h4"
      className=""
      inverted={false}
      textAlign={null}
    >
      <h4
        className="ui header"
      >
        2 My god I love Cliff Richard
      </h4>
    </Header>
  </Heading>
  <FlexContainer
    alignContent={null}
    alignItems={null}
    flexDirection={null}
    flexWrap="wrap"
    justifyContent={null}
  >
    <div
      className="flex-container"
      style={
        Object {
          "alignContent": null,
          "alignItems": null,
          "display": "flex",
          "flexDirection": null,
          "flexGrow": "1",
          "flexWrap": "wrap",
          "height": "100%",
          "justifyContent": null,
        }
      }
    >
      <PropertySearchResult
        key="0"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
      <PropertySearchResult
        key="1"
      >
        <div>
          Property search result
        </div>
      </PropertySearchResult>
    </div>
  </FlexContainer>
</PropertySearchResultList>
`;
